cli/tools/doc.rs

```diff
@@ -37,52 +85,30 @@ pub async fn print_docs(
   let capturing_parser =
     CapturingModuleParser::new(Some(&source_parser), &store);
 
-  let mut doc_nodes = match doc_flags.source_files {
+  let doc_nodes_by_url = match doc_flags.source_files {
     DocSourceFileFlag::Builtin => {
-      let source_file_specifier =
-        ModuleSpecifier::parse("internal://lib.deno.d.ts").unwrap();
-      let content = get_types_declaration_file_text(cli_options.unstable());
-      let mut loader = deno_graph::source::MemoryLoader::new(
-        vec![(
-          source_file_specifier.to_string(),
-          deno_graph::source::Source::Module {
-            specifier: source_file_specifier.to_string(),
-            content,
-            maybe_headers: None,
-          },
-        )],
-        Vec::new(),
-      );
-      let mut graph = deno_graph::ModuleGraph::new(GraphKind::TypesOnly);
-      graph
-        .build(
-          vec![source_file_specifier.clone()],
-          &mut loader,
-          deno_graph::BuildOptions {
-            module_analyzer: Some(&analyzer),
-            ..Default::default()
-          },
-        )
-        .await;
-      let doc_parser = doc::DocParser::new(
-        &graph,
+      generate_doc_nodes_for_builtin_types(
+        doc_flags.clone(),
+        cli_options,
         capturing_parser,
-        doc::DocParserOptions {
-          private: doc_flags.private,
-          diagnostics: false,
-        },
-      )?;
-      doc_parser.parse_module(&source_file_specifier)?.definitions
+        &analyzer,
+      )
+      .await?
     }
-    DocSourceFileFlag::Paths(source_files) => {
+    DocSourceFileFlag::Paths(ref source_files) => {
       let module_graph_builder = factory.module_graph_builder().await?;
       let maybe_lockfile = factory.maybe_lockfile();
 
+      let expanded_globs =
+        expand_globs(source_files.iter().map(PathBuf::from).collect())?;
```

Comment:
I'm not sure about doing globbing here. Do we support globs in any other arguments? Maybe we should move this to a separate PR.

Reply:
Yes, globs are supported in other subcommands as well